src/code_path.ml

```diff
@@ -0,0 +1,34 @@
+type t =
+  { file_path : string
+  ; main_module_name : string
+  ; val_path : string list
+  }
+
+let top_level ~file_path =
+  let main_module_name =
+    file_path
+    |> Filename.basename
+    |> Filename.remove_extension
+    |> String.capitalize_ascii
+  in
+  {file_path; main_module_name; val_path = []}
+
+let file_path t = t.file_path
+let main_module_name t = t.main_module_name
+let val_path t = List.rev t.val_path
+
+let fully_qualified_path t = 
+  String.concat "." (t.main_module_name :: (val_path t))
+
+let enter mod_or_val_name t =
+  {t with val_path = mod_or_val_name :: t.val_path}
+
+let to_string_path t =
+  let val_path = val_path t in
+  let sub_module_path =
+    let is_module_name s = s <> "" && Base.(Char.is_uppercase (String.get s 0)) in
```

Comment:
We open `Import` in all modules, so we should do the same here. It's easier to reason about the code if everything uses the same stdlib.

Reply:
Good point, I'll add that!

Reply:
That triggers a warning because of the use of stdlib's `Filename`. Should I add `core_kernel` as a dependency or simply use `Caml.Filename` here?

Reply:
Let's just use `Caml.Filename`